src/test/regress/expected/multi_with.out

```diff
@@ -0,0 +1,870 @@
+  WHERE
+    users_table.user_id = events_table.value_2 AND event_type IN (4, 5, 6)
+)
+SELECT non_colocated_1.user_id, count(*) 
+FROM
+  non_colocated_1, non_colocated_2
+WHERE
+  non_colocated_1.user_id = non_colocated_2.user_id
+GROUP BY
+  1
+ORDER BY
+  2 DESC, 1;
+ user_id | count 
+---------+-------
+       2 | 60588
+       4 | 21160
+       3 | 13005
+       5 | 10140
+       1 |  4802
+(5 rows)
+
+SET citus.task_executor_type TO 'real-time';
+-- Two colocated CTE under a non-colocated join 
+WITH colocated_1 AS (
+  SELECT 
+    users_table.user_id, events_table.value_2
+  FROM
+    users_table, events_table
+  WHERE
+    users_table.user_id = events_table.user_id AND event_type IN (1, 2, 3)
+),
+colocated_2 AS (
+  SELECT 
+    users_table.user_id, events_table.value_2
+  FROM
+    users_table, events_table
+  WHERE
+    users_table.user_id = events_table.user_id AND event_type IN (4, 5, 6)
+)
+SELECT colocated_1.user_id, count(*) 
+FROM
+  colocated_1, colocated_2
+WHERE
+  colocated_1.value_2 = colocated_2.value_2
+GROUP BY
+  1
+ORDER BY
+  2 DESC, 1;
+ user_id | count 
+---------+-------
+       3 | 26792
+       4 | 25024
+       5 | 22724
+       2 | 22554
+       6 |  5720
+       1 |  5593
+(6 rows)
+
+-- CTE in WHERE basic
+SELECT
+  count(*)
+FROM
+  events_table
+WHERE
+  event_type
+IN
+  (WITH events AS (
+    SELECT 
+      event_type 
+    FROM 
+      events_table 
+    WHERE user_id < 5 
+    GROUP BY 
+      1 
+    ORDER BY 
+      1)
+    SELECT * FROM events LIMIT 10
+  );
+ count 
+-------
+   101
+(1 row)
+
+SET citus.task_executor_type = 'task-tracker';
+-- CTE with non-colocated join in WHERE
+SELECT
+  count(*)
+FROM
+  events_table
+WHERE
+  user_id IN
+    (WITH users AS
+      (SELECT
+          events_table.user_id
+        FROM
+          events_table, users_table
+        WHERE
+          events_table.value_2 = users_table.value_2
+        GROUP BY 
+          1
+        ORDER BY
+          1
+      )
+      SELECT * FROM users LIMIT 10
+    );
+ count 
+-------
+   101
+(1 row)
+
+-- this query doesn't make much sense though but tests colocated join and non-colocated join together
+WITH users_events AS (
+  WITH colocated_join AS (
+    SELECT
+      users_table.user_id as uid, events_table.event_type
+    FROM
+        users_table
+      join
+        events_table
+      on (users_table.user_id = events_table.user_id)
+    WHERE
+      users_table.value_2 IN (1, 2)
+  ),
+  non_colocated_join AS (
+    SELECT
+      users_table.user_id as n_uid, events_table.event_type
+    FROM
+        users_table
+      join
+        events_table
+      on (users_table.value_2 = events_table.value_2)
+  )
+  SELECT
+    uid, colocated_join.event_type
+  FROM
+      colocated_join
+    INNER join
+      non_colocated_join
+    on (colocated_join.event_type = non_colocated_join.event_type)
+)
+SELECT
+  sum(uid), sum(event_type)
+FROM
+  users_events;
+  sum   |  sum   
+--------+--------
+ 972702 | 648967
```

Comment:
go Citus!

Reply:
if you find queries that error out, could you also add those to the regression tests?

Reply:
Sure!